src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
```

Comment:
> this will require us to call GetAscii twice. may be this ok but did we think we can allocate extra memory instead of calling GetAscii twice?

Why is this better?  Do we have measurements to demonstrate that?  We're effectively comparing:
- 0-extra allocation and two interop calls
- up-to-N-extra-allocation and likely one interop call (but maybe two)